scipy/signal/_peak_finding.py

```diff
@@ -233,6 +241,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
+        calculation significantly for large vectors.
+    ret_bases : bool
+        If true, the peaks' left and right bases are returned. See returned
+        arguments below.
+
+    Returns
+    -------
+    prominences : ndarray
+        The calculated prominences for each peak in `peaks`.
+    left_bases, right_bases : ndarray, returned if ret_bases is true
+        The peaks base positions in `vector` to the left and right of each peak.
+        The higher base of each pair is a peak's lowest contour line (see Notes
+        for a more details).
+
+    See Also
+    --------
+    find_peaks, peak_widths
+
+    Notes
+    -----
+    Bellow the strategy to calculate a peak's prominence is summarized:
+
+    * Extend a horizontal line from the current peak to the left and right until
+      the line either reaches the window end (see `wlen`) or intersects the
+      signal again at the slope of a higher peak.
+    * On each side find the minimal signal value within the interval defined
+      above. The position of these two points can be optionally returned
+      (see `ret_bases`).
+    * The higher one of the two points marks the peak's lowest contour line. The
+      prominence can then be calculated as the vertical difference between the
+      peaks height itself and its lowest contour line.
+
+    .. versionadded:: 1.1.0
+
+    References
+    ----------
+    .. [1] Wikipedia Article for Topograhpic Prominence:
+       https://en.wikipedia.org/wiki/Topographic_prominence
+
+    Examples
+    --------
+    Create signal test signal with two overlayed harmonic functions
+
+    >>> x = np.linspace(0, 6 * np.pi, 1000)
+    >>> vector = np.sin(x) + 0.6 * np.sin(2.6 * x)
+
+    Find all peaks and calculate prominences
+
+    >>> from scipy import signal
+    >>> peaks = signal.find_peaks(vector)
+    >>> prominences = signal.peak_prominences(vector, peaks)
+    >>> prominences
+    array([ 1.24159486,  0.47840168,  0.28470524,  3.10716793,  0.284603  ,
+            0.47822491,  2.48340261,  0.47822491])
+
+    Calculate the height of each peak's contour line
+
+    >>> contour_heights = vector[peaks] - prominences
+
+    Plot signal, each peak and its prominence
+
+    >>> import matplotlib.pyplot as plt
+    >>> plt.plot(vector)
+    >>> plt.plot(peaks, vector[peaks], "x")
+    >>> plt.vlines(peaks, ymin=contour_heights, ymax=vector[peaks])
+    """
+    vector = np.asarray(vector)
+    peaks = np.asarray(peaks)
+    if peaks.size == 0:
+        # Handle empty peaks
+        if ret_bases:
+            return np.array([]), np.array([]), np.array([])
+        else:
+            return np.array([])
+
+    if vector.ndim != 1 or peaks.ndim != 1:
+        raise ValueError('vector and peaks must have exactly one dimension')
+    if peaks.max() >= vector.size:
+        raise ValueError('peak position exceeds the range of vector')
+    if not np.issubdtype(peaks.dtype, int):
```

Comment:
Some of the builds / test fail with the the warning:
> FutureWarning: Conversion of the second argument of issubdtype from `int` to `np.signedinteger` is deprecated. In future, it will be treated as `np.int64 == np.dtype(int).type`.

I have a feeling that using `np.issubdtype(peaks.dtype, np.integer)` would be the fix for this but I can't reproduce this warning on my system. Any insights before I blindly push this?